lib/foreman_rh_cloud/engine.rb

```diff
@@ -122,12 +122,24 @@ class Engine < ::Rails::Engine
     end
 
     config.to_prepare do
-      RemoteExecutionFeature.register(
-        :rh_cloud_remediate_hosts,
-        N_('Apply Insights recommendations'),
-        description: N_('Run remediation playbook generated by Insights'),
-        host_action_button: false
-      )
+      # skip database manipulations while tables do not exist, like in migrations
+      if ActiveRecord::Base.connection.data_source_exists? ForemanTasks::Task.table_name
+        RemoteExecutionFeature.register(
+          :rh_cloud_remediate_hosts,
+          N_('Apply Insights recommendations'),
+          description: N_('Run remediation playbook generated by Insights'),
+          host_action_button: false
+        )
+
+        unless ForemanTasks::RecurringLogic.joins(:tasks).merge(
+          ForemanTasks::Task.where(label: 'InventorySync::Async::InventoryScheduledSync')
+        ).exists?
+          recurring_logic = ForemanTasks::RecurringLogic.new_from_cronline("0 0 * * *")
+          recurring_logic.save!
+          recurring_logic.start(InventorySync::Async::InventoryScheduledSync)
+        end
+      end
+    rescue ActiveRecord::NoDatabaseError
```

Comment:
Do we need the rescue if we're checking if the table exists or does the check actually raise this?

Reply:
Yes, we need. I have tried it. The rescue is helping to run `rake db:create` and the table existence test helps to run `rake db:migrate`